### PR TITLE
[Merged by Bors] - Fix misleading error

### PR DESCRIPTION
**Pull request by poszu**
## Motivation
Users often get confused by the log message when the nipost_challenge.bin doesn't exist. This is a normal situation, yet the log includes "errormsg".

## Changes
Don't log an error in case the file is not found.

Additionally, includes a fix for a bug discovered during testing. The code panicked when the file was too small (for example empty).

## Test Plan
- UT
- [x] manual test

### CODE REVIEW

**Review by dshulyak — APPROVED** [review]

File: activation/nipost_state_test.go [hunk 1]
```diff
@@ -58,6 +58,16 @@ func TestReadCRC(t *testing.T) {
...
+			name:   "file too short",
+			data:   []byte("123"),
+			crc:    []byte{},
+			errMsg: "too small",
+		}, {
+			name:   "file empty",
+			data:   []byte(""),
+			crc:    []byte{0xC6, 0x10, 0x72, 0xDF, 0x52, 0xD7, 0x74, 0x0E},
+			errMsg: "wrong checksum 0xC61072DF52D7740E, computed 0x0",
 		},
```
Comment (fasmat) [inline activation/nipost_state_test.go:71]: I'm confused by this test. The CRC64 of an empty file should be `0x0000000000000000` and if the file only contains those bytes it will pass reading & CRC validation

File: activation/activation.go [hunk 2]
```diff
@@ -527,9 +527,11 @@ func (b *Builder) PublishActivationTx(ctx context.Context) error {
 
 	challenge, err := b.loadChallenge()
 	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			logger.With().Error("failed to load atx challenge", log.Err(err))
+		}
```
Comment (fasmat) [inline activation/activation.go]: I would change it to debug logging instead of only logging it when there is no file.

If it is another error (e.g. CRC validation fails) it will log the error and start building a new challenge just like if no file was found. If it would be an error I would expect a different behaviour (e.g. stopping smeshing)

File: activation/nipost_state_test.go (hunk 1 above)
Comment (poszu) [reply activation/nipost_state_test.go:71]: What is confusing? It's exactly as you said. CRC of nothing is 0. The test file contains a wrong CRC and no data hence it should fail with "wrong checksum 0xC61072DF52D7740E, computed 0x0" message.

File: activation/activation.go (hunk 2 above)
Comment (poszu) [reply activation/activation.go]: I don't understand. This branch will be entered when the file exists and it is corrupted. The change is to not log when the file simply doesn't exist.

File: activation/nipost_state_test.go (hunk 1 above)
Comment (fasmat) [reply activation/nipost_state_test.go:71]: but isn't this just a repetition of the "badCRC" testcase?

File: activation/activation.go (hunk 2 above)
Comment (fasmat) [reply activation/activation.go]: OK, let me rephrase: This is not an error, the node doesn't behave differently because it couldn't load the ATX challenge (it does the same no matter if there is no file or a corrupted file). Imo the appropriate logging level is Warn or Debug, because any error doesn't change anything in the flow of the program.

Comment (poszu) [reply activation/activation.go]: For sure it must not be a `Debug`. A corrupted nipost_challenge.bin is a serious thing and the user must be made aware of it. Whether Error or Warn is better, IDK. What are the guidelines here?

File: activation/nipost_state_test.go (hunk 1 above)
Comment (poszu) [reply activation/nipost_state_test.go:71]: No, it's an edge case when the input is empty.

**Comment by poszu** [comment]
bors merge

File: activation/activation.go (hunk 2 above)
Comment (fasmat) [reply activation/activation.go]: > A corrupted nipost_challenge.bin is a serious thing and the user must be made aware of it.

But then just logging it and deleting / overwriting it with a new challenge is probably wrong? Maybe this is serious enough to have the node stop smeshing with an error that is logged and emitted as event.

> What are the guidelines here?

We don't have a logging strategy yet. Here are my 2 cents: anything we log as `Error` must be severe enough to cause the node to deviate from its normal behavior, `Warn` logs on the other hand are issues the user should be informed about but the node can continue to operate as normal. `Fatal` are errors the node can't deal with and has to exit (should ideally never be necessary).

File: activation/nipost_state_test.go (hunk 1 above)
Comment (fasmat) [reply activation/nipost_state_test.go:71]: Again back to my first comment: an empty input is not an error, it has a valid checksum. This test does not check if empty input is handled correctly (that would be "" and 0x0), it checks if a wrong checksum is detected (as does the "badCRC" testcase)

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
flake in `TestNodesUsingDifferentPoets`, unrelated to this change. 

bors merge

**Comment by poszu** [comment]
bors merge
